tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -17,38 +17,39 @@ internal static class DatabaseMonitoringPropagator
         private const string SqlCommentVersion = "ddpv";
         private const string SqlCommentEnv = "dde";
 
-        internal static string PropagateSpanData(DbmPropagationLevel propagationStyle, string configuredServiceName, SpanContext context)
+        internal static string PropagateSpanData(DbmPropagationLevel propagationStyle, string configuredServiceName, SpanContext context, IntegrationId integrationId)
         {
-            if (propagationStyle != DbmPropagationLevel.Service && propagationStyle != DbmPropagationLevel.Full)
+            if ((integrationId is IntegrationId.MySql or IntegrationId.Npgsql or IntegrationId.SqlClient) &&
+                (propagationStyle is DbmPropagationLevel.Service or DbmPropagationLevel.Full))
             {
-                return string.Empty;
+                var propagatorSringBuilder = StringBuilderCache.Acquire(StringBuilderCache.MaxBuilderSize);
+                propagatorSringBuilder.Append($"/*{SqlCommentSpanService}='{Uri.EscapeDataString(context.ServiceName)}'");
+
+                if (context.TraceContext?.Environment is { } envTag)
+                {
+                    propagatorSringBuilder.Append($",{SqlCommentEnv}='{Uri.EscapeDataString(envTag)}'");
+                }
+
+                propagatorSringBuilder.Append($",{SqlCommentRootService}='{Uri.EscapeDataString(configuredServiceName)}'");
+
+                if (context.TraceContext?.ServiceVersion is { } versionTag)
+                {
+                    propagatorSringBuilder.Append($",{SqlCommentVersion}='{Uri.EscapeDataString(versionTag)}'");
+                }
+
+                if (propagationStyle == DbmPropagationLevel.Full && integrationId is not IntegrationId.SqlClient)
```

Comment:
We don't want to inject the W3C `traceparent` header in SQL Server queries?

Reply:
Nah not yet because of the way it generates new plans for any change to a query, in the [RFC ](https://docs.google.com/document/d/1hlp3CuIRE-os2zYgc0wYOS8kPtCKqlrGL7aepbHHDaU/edit#heading=h.r2lhua4oseij)here they added more details about it, so for SqlServer we'll only support adding the simple tags for now.

Reply:
Got it! Maybe add a comment here to explain that?